rtt/internal/BindStorage.hpp

```diff
@@ -335,10 +337,10 @@ namespace RTT
             void store(arg1_type t1) { a1(t1); }
             void exec() {
 #ifdef ORO_SIGNALLING_OPERATIONS
-                if (msig) (*msig)(a1.get());
+                if (msig) (*msig)(a1);
 #endif
                 if (mmeth)
-                    retv.exec( boost::bind(mmeth, boost::ref(a1.get()) ) );
+                    retv.exec( boost::bind(mmeth, a1 ) );
```

Comment:
To be checked: Does `boost::bind` actually copies the `AStore<T>` instance and indirectly the argument to the functor object or does it invoke `operator T&()` here and only stores a reference?

Reply:
Seems like [boost::bind]() is always copying the arguments itself, and only invokes the conversion operator at the time the functor is actually called: https://stackoverflow.com/questions/11255144/why-does-boostbind-store-arguments-of-the-type-passed-in-rather-than-of-the-ty

In this case the patch in a5857b2aab16144ed5ddb314209529de0d06cc9c (this PR) and a2ccd7961975a80f4befd699a7edc8b029085cea (in `toolchain-2.9`) is actually wrong. The `.get()` can be removed, but the `boost::ref()` call should have been kept.